### PR TITLE
Use Patch to update Nodes and LoadBalancer status

### PR DESCRIPTION
**Pull request by zoumo**
feat: use json-patch to update nodes and lb status
feat: change label value format from "namespace_name" to "namespace.name"
fix: clean up ingresses and configmaps when lb has been deleted

### CODE REVIEW

**Comment by zoumo** [comment]
@kdada 
/cc @ddysher

**Comment by kdada** [comment]
Split to more commits.

File: pkg/apis/networking/v1alpha1/types.go [hunk 1]
```diff
@@ -172,9 +172,9 @@ type AzureProvider struct {
 // LoadBalancerStatus represents the current status of a LoadBalancer
 type LoadBalancerStatus struct {
 	// +optional
-	ProxyStatus ProxyStatus `json:"proxyStatus"`
+	ProxyStatus ProxyStatus `json:"proxyStatus,omitempty"`
```
Comment (kdada) [inline pkg/apis/networking/v1alpha1/types.go]: Use pointer or it can't be omitted.

File: pkg/apis/networking/v1alpha1/types.go [hunk 2]
```diff
@@ -172,9 +172,9 @@ type AzureProvider struct {
...
 	// +optional
-	ProvidersStatuses ProvidersStatuses `json:"providersStatuses"`
+	ProvidersStatuses ProvidersStatuses `json:"providersStatuses,omitempty"`
```
Comment (kdada) [inline pkg/apis/networking/v1alpha1/types.go]: ditto

**Comment by kdada** [comment]
LGTM

**Pull request merged**